### ISSUE TITLE
SearchForFunctionInCode test failing due to URL encoding

### ISSUE THREAD

**Issue by shiftkey** [octokit/octokit.net#474]
Request:

```
GET https://api.github.com/search/code?page=1&per_page=100&order=desc&q=addClass%2Brepo%3Ajquery%2Fjquery HTTP/1.1
Accept: application/vnd.github.v3+json; charset=utf-8
User-Agent: OctokitTests (Win32NT 6.3.9600; amd64; en-US; Octokit 0.3.2)
Authorization: Basic {token}
Host: api.github.com
Accept-Encoding: gzip, deflate
Connection: Keep-Alive
```

Response:

```
HTTP/1.1 200 OK
Server: GitHub.com
Date: Wed, 30 Apr 2014 07:09:55 GMT
Content-Type: application/json; charset=utf-8
Status: 200 OK
X-RateLimit-Limit: 30
X-RateLimit-Remaining: 29
X-RateLimit-Reset: 1398841855
Cache-Control: no-cache
X-GitHub-Media-Type: github.v3; format=json
Link: <https://api.github.com/search/code?page=0&per_page=100&order=desc&q=addClass%2Brepo%3Ajquery%2Fjquery>; rel="last"
X-XSS-Protection: 1; mode=block
X-Frame-Options: deny
Content-Security-Policy: default-src 'none'
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Access-Control-Allow-Origin: *
X-GitHub-Request-Id: 017EAAA5:04FE:E25761:5360A1C2
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
Vary: Accept-Encoding
X-Served-By: c436b2b44345c72ff906059f604991e7
Content-Length: 55

{"total_count":0,"incomplete_results":false,"items":[]}
```

TL;DR: we need a way to whitelist certain querystring parameters, or have the backend support decoding parameters correctly

```
Docs: q=addClass+repo:jquery/jquery
Client: q=addClass%2Brepo%3Ajquery%2Fjquery
```
